app/src/main/java/com/sopt/now/compose/feature/common/base/BaseFactory.kt

```diff
@@ -0,0 +1,11 @@
+package com.sopt.now.compose.feature.common.base
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class BaseFactory<T>(
+    private val creator: () -> T,
+) : ViewModelProvider.Factory where T : ViewModel {
```

Comment:
오 where 키워드를 아시는 걸로 보아 코틀린에 관심이 많으시군요 😸  

하지만, where 를 사용하신 이유가 있을까요?  

```suggestion
class BaseFactory<T: ViewModel>(
    private val creator: () -> T,
) : ViewModelProvider.Factory  {
```

Reply:
좋은 지적 감사합니다!! 아직 전 제너릭이랑 안친한 것 같아요 🥲

Reply:
제네릭이 많이 어렵지만
Kotlin/Java 개발자라면 필수적으로 학습해야할 문법이라고 생각해요!!

만약, 학습하시다가 궁금한 점이 생기시면
Discord DM 으로 언제든 물어봐주세요😉